src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -533,7 +533,7 @@ void network::set_output_memory(const primitive_id& id, memory::ptr mem_new) {
     }
 
     for (auto& prim : o_iter->second) {
-        prim->set_output_memory(eng.reinterpret_buffer(*mem_new, prim->output_memory().get_layout()), false);
+        prim->set_output_memory(eng.reinterpret_buffer(*mem_new, prim->get_node_output_layout()), false);
```

Comment:
What if node has dynamic layout? I think we may need to set out memory for primitive even for dynamic cases (for cases with known upper-bound maybe)